### DISCUSSION TITLE
Winter is coming => Floorheating pump only when there is a request for heat

### DISCUSSION THREAD

**Discussion by mhoogenbosch** [emsesp/EMS-ESP32#2069] [General]
I've had the for a decent period now. Besides some show and tell on the dashboard of Home Assistant, I've not really been using it. I'm still looking for a 'better' way of utilizing the possibilities.

Now winter is coming, and my floor heater pump has been turned off for a couple of months now (with an interval of once per 24hr). I'd really like it to turn only on when there is an ask for heat. I've seen people using a temperature sensor for that, but I'd like to use the boiler. Furthermore, I have Home Assistant in place, as goes for a smart plug with the pump.

We have a moduline 400 thermostat, but with separate room temperature sensors which are analog and aren't seen by EMS-ESP (obiously). The boiler is a topline Nefit model. 

Which sensor should I use to create the automation with? I thought it would make sense to use binary_sensor.boiler_heatingactive, but it 'flipps' a lot. 

![image](https://github.com/user-attachments/assets/08d82116-ba33-4739-9352-5b64bfa1cdbc)

Support information:

[emsesp_system_undefined.txt](https://github.com/user-attachments/files/17200327/emsesp_system_undefined.txt)

I have tried turning on a room thermostat, but it doens't make the heatingactive to turn on instantly. It takes a while, and I don't know if it is actually related to each other.

**Comment by proddy** [comment]
Using the boiler directly is tricky. There is not one single on/off parameter to change. Take a look at https://emsesp.org/FAQ/#can-ems-esp-simulate-a-thermostat for some notes on this. Also modifying the boiler directly may interfere with your Moduline 400 and the thermostat tries to be smart and turns it off. 

It would be easier to get those external analog temp sensor data into HA and then use that data to drive EMS-ESP Remote 'dummy' thermostats.
